src/Npgsql/Internal/NpgsqlConnector.cs

```diff
@@ -535,7 +535,11 @@ internal async Task Open(NpgsqlTimeout timeout, bool async, CancellationToken ca
         {
             await conn.RawOpen(sslMode, timeout, async, cancellationToken, isFirstAttempt);
 
+#if NET5_0_OR_GREATER
+            var username = await conn.GetUsernameAsync();
```

Comment:
Can't we get here on .net 6 with a sync `Open`, and as such `GetUsernameAsync` should also be sync in that case?